### PR TITLE
Refactor regressions

### PR DESCRIPTION
**Pull request by lintool**
+ "threads" and "batch-size" parameters:
  + consistent formatting
  + settable externally in code (rather than hard-coded in yaml)
+ better logging information for running times
+ run_jobs_with_load.py won't spawn too many concurrent jobs, configurable parameter.

### CODE REVIEW

**Comment by lintool** [comment]
@jasper-xian @ronakice this PR is ready for review now.

Just a quick sanity check is sufficient, I've run the following and they pass; I'll run again from scratch after merge.

```
nohup python -m pyserini.2cr.msmarco --collection v1-passage --all --directory runs/ --display-commands >& logs/log.msmarco-v1-passage &
nohup python -m pyserini.2cr.msmarco --collection v1-doc     --all --directory runs/ --display-commands >& logs/log.msmarco-v1-doc &
nohup python -m pyserini.2cr.msmarco --collection v2-passage --all --directory runs/ --display-commands >& logs/log.msmarco-v2-passage &
nohup python -m pyserini.2cr.msmarco --collection v2-doc     --all --directory runs/ --display-commands >& logs/log.msmarco-v2-doc &
nohup python -m pyserini.2cr.miracl --all --directory runs/ --display-commands >& logs/log.miracl &
nohup python -m pyserini.2cr.mrtydi --all --directory runs/ --display-commands >& logs/log.mrtydi &
nohup python -m pyserini.2cr.beir   --all --directory runs/ --display-commands >& logs/log.beir &
nohup python -m pyserini.2cr.odqa   --all --directory runs/ --topic tqa --display-commands >& logs/log.odqa.tqa &
nohup python -m pyserini.2cr.odqa   --all --directory runs/ --topic nq  --display-commands >& logs/log.odqa.nq &
```

File: pyserini/2cr/beir.py
Comment (jasper-xian) [inline pyserini/2cr/beir.py:31]: I think it might make more sense to move these constants into `pyserini/2cr/base` since they're reused a bunch. Your call though

Comment (jasper-xian) [reply pyserini/2cr/beir.py:31]: Otherwise, this looks good to me!

Comment (lintool) [reply pyserini/2cr/beir.py:31]: I was thinking that I might want the ability to control on a per-regression basis, so I'll keep separate?

Please +1 the review so I can merge? Thanks!

**Review by jasper-xian — APPROVED** [review]

**Pull request merged**
